### ISSUE TITLE
Rule Change: `no-extra-parens` should have option to keep parenthesis for spread operator with ternary inside

### ISSUE THREAD

**Issue by mvorisek** [eslint/eslint#16626]
### What rule do you want to change?

no-extra-parens

### What change to do you want to make?

It was reported already in https://github.com/eslint/eslint/issues/13691. It is not a bug, but
```...(a ? b : c)```
is much more readable than without the parenthesis
```...a ? b : c```

I would say
```...(a ? b : c)```
is generally preferred CS, thus the rule should support it. Currently the only option is to disable the rule (globally or per each usage), but the rule is otherwise very powerfull, so I belive it deserves an extra config option.

### What does the rule currently do for this code?

remove parenthesis for spread operator in all cases, even for ternary/null coalescing operator inside

### What will the rule do after it's changed?

with the new requested option, keep parenthesis for spread operator with ternary/null coalescing operator inside

**Comment by mdjermanovic** [comment]
Hi @mvorisek, thanks for the issue!

This is a style preference in a stylistic rule so it falls under our [Stylistic Rule Updates](https://github.com/eslint/eslint#stylistic-rule-updates) policy. Also, this rule was always intended to report all extra parentheses and the only exceptions were for cases where it was necessary to avoid conflicts with other rules (e.g., `conditionalAssign` option was added only to avoid conflicts with the `no-cond-assign` rule).

You can always make a custom rule based on the core rule and make it behave the way you want. This comment from another issue with the same request might be helpful: https://github.com/eslint/eslint/issues/13409#issuecomment-643259063

**Comment by mvorisek** [comment]
I understand the decision and thank you for https://github.com/eslint/eslint/issues/13409#issuecomment-643259063.

Would you please be so kind and advise also how to extend/add the custom rule into `.eslintrc.js` directly (ie. without plugin in a seperate file)

**Comment by mdjermanovic** [comment]
> how to extend/add the custom rule into `.eslintrc.js` directly (ie. without plugin in a seperate file)

It isn't possible to add rule implementations directly into `.eslintrc.js` config files. You'll need to create a plugin, and that would be at least one separate file. If you'd like to avoid publishing the plugin, check how we're using [`eslint-plugin-internal-rules`](https://github.com/eslint/eslint/blob/7276fe5776f03fb90e575ed63a9b1a6766993e42/package.json#L112) in this project.

It is possible to add rule implementations directly into [`eslint.config.js`](https://eslint.org/docs/latest/user-guide/configuring/configuration-files-new) config files (note that this new config format is still considered experimental).

**Comment by mvorisek** [comment]
In case anyone wants to keep/allow the parenthesis around ternary operator when used in spread or another ternary operator, here is a working solution: https://github.com/fomantic/Fomantic-UI/pull/2589/commits/798ecbca4894a70a64489574938982eb5a316a7b
